filament/src/RenderPass.h

```diff
@@ -208,27 +209,28 @@ class RenderPass {
         return boolish ? std::numeric_limits<uint64_t>::max() : uint64_t(0);
     }
 
-    struct PrimitiveInfo { // 24 bytes
-        FMaterialInstance const* mi = nullptr;                          // 8 bytes (4)
-        backend::Handle<backend::HwRenderPrimitive> primitiveHandle;    // 4 bytes
-        backend::RasterState rasterState;                               // 4 bytes
-        uint16_t index = 0;                                             // 2 bytes
-        Variant materialVariant;                                        // 1 byte
-        uint8_t reserved[13 - sizeof(void*)] = {};                      // 5 byte (9)
+    struct PrimitiveInfo { // 32 bytes
+        FMaterialInstance const* mi = nullptr;                            // 8 bytes (4)
+        PerRenderPrimitiveMorphing const* primitiveMorphing = nullptr;    // 8 bytes (4)
```

Comment:
This is a big increase per primitive for something that won't be used by most primitives

Reply:
Yes I know but I couldn't find other ways. Each primitive can have a morphing target. I thought that putting morphing info into HwRenderPrimitive but it's not a good idea. Because PerRenderPrimitiveMorphing has the sampling binding location info and the driver layer shouldn't know about this. If you have good ideas, please let me know.

Reply:
I found a way to pack the size of command by 32 bytes. We don't have to worry about the big increasement per the command.